core/kt-osrd-utils/src/main/kotlin/fr/sncf/osrd/utils/json/UnitJsonAdapters.kt

```diff
@@ -49,6 +47,26 @@ class OffsetAdapter<T> : JsonAdapter<Offset<T>?>() {
     }
 }
 
+/**
+ * Utility class, used to put Speeds directly in json-adaptable classes. A value of type `double`
+ * will be expected, representing meters per second.
+ */
+class SpeedAdapter : JsonAdapter<Speed?>() {
+    @FromJson
+    override fun fromJson(reader: JsonReader): Speed? {
+        if (reader.peek() == JsonReader.Token.NULL) {
+            reader.skipValue()
+            return null
+        }
+        return reader.nextDouble().metersPerSecond
+    }
+
+    @ToJson
+    override fun toJson(writer: JsonWriter, value: Speed?) {
+        writer.value(if (value != null) Speed.toMetersPerSecond(value) else null)
```

Comment:
This choice is debatable: serialize in meters per second (float), while internal value is in millimeters per second (unsigned int).
This eases the use in front.

Reply:
Looks fine for me, these classes are used to decorrelate the internal representations from the ones in the api